# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: caps/OriginAttributes.h
# Commit: 08acd8256071
# Full Hash: 08acd8256071ad86345dec3fa66767635b07a4b1
# Author: Tim Huang <tihuang@mozilla.com>
# Date: 2025-01-09 04:16:57
# Regressor Bug: 1915355
# File Overlap Count: 1
# Description:
#   Bug 1915355 - Part 4: Implement the third-party cookie blocking excpetion in CookieService. r=cookie-reviewers,baku,valentin
#   
#   The patch integrates the third-party cookie blocking excpetion in
#   CookieService.
#   
# ==============================================================================

diff -r 3ae9463ee988 -r 08acd8256071 caps/OriginAttributes.h
--- a/caps/OriginAttributes.h	Wed Jan 08 19:25:58 2025 +0000
+++ b/caps/OriginAttributes.h	Wed Jan 08 19:25:59 2025 +0000
@@ -141,6 +141,11 @@
                                 nsAString& outScheme, nsAString& outBaseDomain,
                                 int32_t& outPort,
                                 bool& outForeignByAncestorContext);
+
+  // Parse a partitionKey and extract the site from it. Returns false if the
+  // partitionKey cannot be parsed because the format is invalid.
+  static bool ExtractSiteFromPartitionKey(const nsAString& aPartitionKey,
+                                          nsAString& aOutSite);
 };
 
 class OriginAttributesPattern : public dom::OriginAttributesPatternDictionary {